lib/std/special/test_runner.zig

```diff
@@ -78,6 +79,12 @@ pub fn main() anyerror!void {
                 progress.log("{s}...SKIP\n", .{test_fn.name});
                 if (progress.terminal == null) std.debug.print("SKIP\n", .{});
             },
+            error.ZigTestFailed => {
+                fail_count += 1;
+                test_node.end();
+                progress.log("{s}...FAIL\n", .{test_fn.name});
+                if (progress.terminal == null) std.debug.print("FAIL\n", .{});
+            },
             else => {
                 progress.log("", .{});
                 return err;
```

Comment:
This should also count as a failure rather than aborting the testing process.

I suggest to make the `std.testing.expectFoo` functions return `error.TestOutputMismatch` or something like that. Then put the "zig test failed" logic in the else prong here, and print the error name rather than "FAIL".